DM-25061: Update unit testing page of developer guide (#402)
lsst-dm/dm_dev_guide: python/testing.rst

`lsst.utils.tests.TestCase.assertFloatsNotEqual`
   Asserts that floating point scalars and/or arrays are not equal.

> Additionally, :ref:`lsst.afw.geom <pipelines:lsst.afw.geom>` and :ref:`lsst.afw.image <pipelines:lsst.afw.image>` provides additional asserts that get loaded into `lsst.utils.tests.TestCase` when the associated module is loaded.

Reviewer: ```suggestion
Additionally, :ref:`lsst.geom <pipelines:lsst.geom>`, :ref:`lsst.afw.geom <pipelines:lsst.afw.geom>`, and :ref:`lsst.afw.image <pipelines:lsst.afw.image>` provide additional asserts that get loaded into `lsst.utils.tests.TestCase` when the associated module is loaded.
```

> These include methods for `Geom (Angles, Pairs, Boxes)`_, and `Images`_, such as:

Reviewer: ```suggestion
These include methods for SpherePoints, `Geom (Angles, Pairs, Boxes)`_, and `Images`_, such as:
```
(Presumably there's some good place to link this to.)

Author: Well, they are all linked within the description of each special assert, just a few lines down.

(I also went with SpherePoints inside those brackets, for what it's worth - they're also provided by just lsst.geom, the tricky odd one out is the special assert for WCS)

> :lmeth:`~lsst.afw.coord.utils.assertSpherePointsAlmostEqual`

Reviewer: ```suggestion
:lmeth:`~lsst.geom.testUtils.assertSpherePointsAlmostEqual`
```
and similarly below.

>    Assert that two sphere points (:lclass:`~lsst.geom.SpherePoint`) are nearly equal (provided by :lmod:`lsst.afw.geom.utils`).

Reviewer: ```suggestion
   Assert that two sphere points (:lclass:`~lsst.geom.SpherePoint`) are nearly equal (provided by :lmod:`lsst.geom.testUtils`).
```
and similarly below.

:lmeth:`~lsst.afw.geom.utils.assertAnglesAlmostEqual`
   Assert that two angles are nearly equal, ignoring wrap differences by default (provided by :lmod:`lsst.afw.geom.utils`).
:lmeth:`~lsst.afw.geom.utils.assertPairsAlmostEqual`
   Assert that two planar pairs (e.g. :lclass:`~lsst.afw.geom.Point2D` or :lclass:`~lsst.afw.geom.Extent2D`) are nearly equal (provided by :lmod:`lsst.afw.geom.utils`).
:lmeth:`~lsst.afw.geom.utils.assertBoxesAlmostEqual`
   Assert that two boxes (:lclass:`~lsst.afw.geom.Box2D` or :lclass:`~lsst.afw.geom.Box2I`) are nearly equal (provided by :lmod:`lsst.afw.geom.utils`).

> :lmeth:`~lsst.afw.image.basicUtils.assertWcsAlmostEqualOverBBox`

Reviewer: ```suggestion
:lmeth:`~lsst.afw.geom.utils.assertWcsAlmostEqualOverBBox`
```

>    Compare :lmeth:`~lsst.afw.image.imageLib.Wcs.pixelToSky` and :lmeth:`~lsst.afw.image.imageLib.Wcs.skyToPixel` for two WCS over a rectangular grid of pixel positions (provided by :lmod:`lsst.afw.image.basicUtils`).

Reviewer: ```suggestion
   Compare :lmeth:`~lsst.afw.image.imageLib.Wcs.pixelToSky` and :lmeth:`~lsst.afw.image.imageLib.Wcs.skyToPixel` for two WCS over a rectangular grid of pixel positions (provided by :lmod:`lsst.afw.geom.utils`).
```
